### PR TITLE
feat(JOML): migrate and clean up

### PR DESCRIPTION
**Pull request by pollend**

### CODE REVIEW

**Comment by jdrueckert** [comment]
Still getting various compile errors, for instance in `/src/main/java/org/terasology/structureTemplates/internal/ui/StructureTemplateEditorScreen.java`

**Comment by jdrueckert** [comment]
@pollend I migrated a big bunch of what's in `StructureTemplates` to align with https://github.com/Terasology/Projectile/pull/13 and https://github.com/Terasology/AdventureAssets/pull/31.
The only thing left according to the compiler is the following:
```
for-each not applicable to expression type
        for (Vector3i pos : region) {
                            ^
  required: array or java.lang.Iterable
  found:    BlockRegion
```

This appears in `GetStructureTemplateBlocksForMidAirEvent`, `StructureSpawnServerSystem`, and `ReplaceWallServerSystem`.
Can you please adjust `BlockRegion` such that it is iterable?
(pinging @skaldarnar as he did the migration of `BlockRegion` IIRC)

**Comment by pollend** [comment]
@jdrueckert  umm, maybe I should do some kind of shorthand
```
    BlockRegionIterable.region(yourRegion).build()
```

**Comment by jdrueckert** [comment]
@pollend Whatever you deem works best :+1:

**Comment by jdrueckert** [comment]
Needs to be merged together with alignments in consuming modules:
- https://github.com/Terasology/ClimateConditions/pull/36
- https://github.com/Terasology/CombatSystem/pull/58
- https://github.com/Terasology/DynamicCities/pull/71
- https://github.com/Terasology/GooeysQuests/pull/57
- https://github.com/Terasology/Lost/pull/25

**Review by jdrueckert — APPROVED** [review]

File: src/main/java/org/terasology/structureTemplates/components/SpawnPrefabsComponent.java [hunk 1]
```diff
@@ -34,7 +34,6 @@
     public static class PrefabToSpawn {
         public Prefab prefab;
         public Vector3i position;
-        public Quat4f rotation = new Quat4f(0,0,0, 0);
+        public Quaternionf rotation = new Quaternionf();
```
Comment (skaldarnar) [inline src/main/java/org/terasology/structureTemplates/components/SpawnPrefabsComponent.java:37]: According to https://joml-ci.github.io/JOML/apidocs/org/joml/Quaternionf.html#Quaternionf-- this is not the same as before ...

**Pull request merged**